Rg analyzer and cleanup of numeric functions headers

hrabbel opened the pull request:
- Deleted the headers utility/NumericFunctions.h and utility/NumericTools.h, as they were not maintained and essentially not used
- Provide a new version of AnalyzerRadiusOfGyration. It analyzes the complete system by default, but lets the user set the groups to be analyzed later. Output is a single file with the time series of the Rg^2, averaged over all groups analyzed. If a more sophisticated grouping of monomers is required for analysis,  one could for example write an own version of this analyzer by inheriting from it, and set the groups differently by overwriting initialize(), while keeping the rest of the functionaliy. One could even use changing groups (e.g. all polymers close to a wall) by resetting the groups from within execute().
- Provide a test for AnalyzerRadiusOfGyration
- Add a new function to ResultFormattingTools, which allows to append to an existing file, or write the data to a file without adding the meta-information header. This is used in AnalyzerRadiusOfGyration to write in regular intervals, such that data is not lost if a job crashes.

Commit:
*Draft of new version of AnalyzerRadiusOfGyration
*Added an additional function to ResultFormattingTools

Commit:
removed the headers NumericFunctions.h and NumericTools.h,
as they are leftovers from previous experiments and essentially unused.

Commit:
removed the calculation of averages from the analyzer, as this can
always be done with other tools later, and it makes the analyzer more
clear and easy to understand.

Commit:
Removed the constructor taking monomer groups as arguments

Instead, the analyzer uses the whole system, and places it
into a monomer group in initialize(), if the groups have not
been set explicitly using the setter function provided.
This has the advantage that one can make an own Rg analyzer class by
inheriting from AnalyzerRadiusOfGyration, and explicitly set
the groups there in the initialize. Otherwise one can use the same
execute and cleanup as provided here.

Commit:
Further simplified AnalyzerRadiusOfGyration to give only the average Rg of the groups as output.

Commit:
* added missing headers in AddMonomer moves
* added test for AnalyzerRadiusOfGyration

Commit:
corrected comment

MartinWenge commented on include/LeMonADE/analyzer/AnalyzerRadiusOfGyration.h:
```diff
@@ -59,227 +44,229 @@ along with LeMonADE.  If not, see <http://www.gnu.org/licenses/>.
-				 const std::vector<MonomerGroup<molecules_type> >& g,
-			  std::string filenamePrefix="",
-			  uint64_t lowerMcsLimit=0);
+			  std::string filename=std::string("Rg2TimeSeries.dat"));
```
I would prefere to change 
std::string filename=std::string("Rg2TimeSeries.dat")
to 
std::string filename("Rg2TimeSeries.dat")
which calls a constructor only, and does not need a = operator call. Of cause this might be optimized by compiler, but it also looks better ;o)

hrabbel replied:
I don't think that would be valid syntax. But I could do ...="Rg2TimeSeries.dat"
if that is better.

MartinWenge replied:
This is a valid copy constructor for sting objects: 
[](http://www.cplusplus.com/reference/string/string/string/)

MartinWenge replied:
You are rigth, I was confusing myself and thought this to be the initialization list...

MartinWenge commented on include/LeMonADE/analyzer/AnalyzerRadiusOfGyration.h:
```diff
@@ -59,227 +44,229 @@ along with LeMonADE.  If not, see <http://www.gnu.org/licenses/>.
+			groups[0].push_back(n);
+	}
+	//initialize the buffer 
+	Rg2TimeSeries.resize(4,std::vector<double>(0));
```
The re-size of `Rg2TimeSeries.resize(4,std::vector<double>(0));` could be done in the initialization list in the constructor like this:
`..., Rg2TimeSeries(4,std::vector<double>()), ...`
This way, the vector is constructed with the proper length and is ready to use before calling initialize. And one does not need a extra call for this :o)

MartinWenge commented on include/LeMonADE/analyzer/AnalyzerRadiusOfGyration.h:
```diff
@@ -59,227 +44,229 @@ along with LeMonADE.  If not, see <http://www.gnu.org/licenses/>.
+	}
+	//otherwise just append the new data
+	else{
+		ResultFormattingTools::appendToResultFile(outputFile,
```
this is a really nice implementation for the output!

MartinWenge commented on include/LeMonADE/utility/ResultFormattingTools.h:
```diff
@@ -146,4 +154,36 @@ void ResultFormattingTools::writeResultFile(std::string filename,const Ingredien
+	
+	if(!file.is_open())
+		throw std::runtime_error("ResultFormattingTools::appendToResultFile(): error opening output file"+filename+"\n");
+	std::stringstream contents;
```
is this `std::stringstream contents;` used somewhere?

hrabbel replied:
nope, forgot to delete it.

Commit:
small changes suggested after code review.

MartinWenge commented on include/LeMonADE/analyzer/AnalyzerRadiusOfGyration.h:
```diff
@@ -59,227 +44,229 @@ along with LeMonADE.  If not, see <http://www.gnu.org/licenses/>.
+	//! Set the number of values, after which the time series is saved to disk
+	void setBufferSize(uint32_t size){bufferSize=size;}
+	//! Set the groups to be analyzed
+	void setMonomerGroups(std::vector<MonomerGroup<molecules_type> > groupVector){groups=groupVector;}
```
Make this function protected to show it is designed to use it in a derived class.

Commit:
made member setMonomerGroups(...) of AnalyzerRadiusOfGyration
protected to make it clear that this function is meant to be used
from derived classes only. Changed the test accordingly.
